agreementbot/api.go

```diff
@@ -365,6 +366,8 @@ func (a *API) agreement(w http.ResponseWriter, r *http.Request) {
 				http.Error(w, "Internal server error", http.StatusInternalServerError)
 			} else if ag == nil {
 				writeInputErr(w, http.StatusBadRequest, &APIUserInputError{Input: "id", Error: "agreement id not found"})
+			} else if err := abstractprotocol.ObscureProposalSecret(&ag.Proposal); err != nil {
+				glog.Error(APIlogString(fmt.Sprintf("failed to obscure secret details, error: %v", err)))
 			} else {
```

Comment:
You need to obscure the agreement.Policy object also. It has the same secretDetails in it. This is only a problem on the agbot side, the agent only has the proposal.